source/retryable-writes/retryable-writes.rst

```diff
@@ -212,24 +212,34 @@ Determining Retryable Errors
 
 When connected to a MongoDB instance that supports retryable writes (versions 3.6+),
 the driver MUST treat all errors with the RetryableWriteError label as retryable.
-This label might be added to an error in a variety of ways:
+This error label can be found in the top-level "errorLabels" field of the error.
+With a WriteConcernError response, the top level document or the
+WriteConcernError document may contain the RetryableWriteError error label.
+
+The RetryableWriteError label might be added to an error in a variety of ways:
 
 When the driver encounters a network error communicating with any server version
 that supports retryable writes, it MUST add a RetryableWriteError label to that
-error.
+error if the MongoClient performing the operation has the retryWrites
+configuration option set to true.
 
 For server versions 4.4 and newer, MongoDB will add a RetryableWriteError label to
 errors or server responses that it considers retryable before returning them to the
 driver. As new server versions are released, the errors that are labeled with the
 RetryableWriteError label may change. When receiving a command result
 with an error from a 4.4+ server that supports retryable writes, the driver
-MUST NOT add a RetryableWriteError label to that error under any condition.   
-Note: With a WriteConcernError response the top level document or the 
-WriteConcernError document may contain the RetryableWriteError error label.
+MUST NOT add a RetryableWriteError label to that error under any condition.
+
+During a retryable write operation on a sharded cluster, mongos may retry the
+operation internally, in which case it will not add a RetryableWriteError label to
+any error that occurs after those internal retries to prevent excessive retrying.
+The driver MUST NOT add a RetryableWriteError label to errors it receives from
+a version 4.4+ mongos.
```

Comment:
I wonder if _this_ behavior isn't covered by our tests though?

Reply:
I believe it is. Every file that tests error label behavior is run on a sharded cluster, and all of those tests include a case where the server does not return an error label and the driver is expected to not retry. Do you think there's something more we need to do to test this case?